curriculum/challenges/english/07-scientific-computing-with-python/learn-interfaces-by-building-an-equation-solver/662bdd364bf2cde1487922a9.md

```diff
@@ -0,0 +1,43 @@
+---
+id: 662bdd364bf2cde1487922a9
+title: Step 5
+challengeType: 20
+dashedName: step-5
+---
+
+# --description--
+
+Unlike other programming languages, Python does not implement interfaces in its core language, but Python standard library allows you to define interfaces in a simple way.
+
+For this project, you'll use utilities from the `abc` module. So, go at the top of your code and import `ABC` from the `abc` module.
+
+You can import a specific object from a module following the import construct `from <module> import <object>`.
+
+# --hints--
+
+You should import `ABC` from the `abc` module.
+
+```js
+({ test: () => assert(runPython(`_Node(_code).has_import("from abc import ABC")`)) })
+```
+
+# --seed--
+
+## --seed-contents--
+
+```py
+--fcc-editable-region--
+
+--fcc-editable-region--
+class Equation:
+    def solve(self): pass
+        
+    def analyze(self): pass
```

Comment:
Is this code orientation intentional?

Reply:
Yeah, but I'm not mentioning it can be done so I'm thinking of restoring the usual indentation.